executor/executor.go

```diff
@@ -166,13 +173,14 @@ func newBaseExecutor(ctx sessionctx.Context, schema *expression.Schema, id fmt.S
 // return a batch of rows, other than a single row in Volcano.
 // NOTE: Executors must call "chk.Reset()" before appending their results to it.
 type Executor interface {
+	This() *baseExecutor
```

Comment:
In the OOP concept, `This` is commonly used to represent the actual object pointer.

Reply:
It's changed to `base` by the suggestion from @coocood

Reply:
OK. But I'm wondering whether this PR is beneficial. 

This PR removed two functions from the `Executor` interface:
* `retTypes() []*types.FieldType`
* `newFirstChunk() *chunk.Chunk`

But it also introduced one function to the `Executor` interface:
* `base() *baseExecutor`

So actually, we removed 1 function indeed.

Reply:
I think `base()` should not be part of the Executor interface either.

Reply:
But in reality, we have to populate some help methods to the `Executor` interface. Not just TiDB, but also other database systems. For example:
* Impala: https://github.com/cloudera/Impala/blob/cdh5-trunk/be/src/exec/exec-node.h